ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/forkchoice/ReadOnlyStore.java

```diff
@@ -32,6 +32,14 @@
 
   UInt64 getGenesisTime();
 
+  /**
+   * Returns the anchor checkpoint from which the chain was started, if such an anchor exists. If
+   * the anchor is missing, the node was started up from genesis.
+   *
+   * @return The anchor if it exists.
+   */
+  Optional<Checkpoint> getAnchor();
```

Comment:
Going forward, this will return genesis if you started up from genesis.  But not sure its worth extra lookups to the cold db to always set a value for older databases.

Reply:
ah I see, that makes sense.